pyam/core.py

```diff
@@ -601,11 +601,10 @@ def categorize(self, name, value, criteria,
                           ('linestyle', linestyle)]:
             if arg:
                 run_control().update({kind: {name: {value: arg}}})
-
         # find all data that matches categorization
-        rows = _apply_criteria(self.data, criteria,
+        rows = _apply_criteria(self._data, criteria,
                                in_range=True, return_test='all')
-        idx = _meta_idx(rows)
+        idx = _make_index(rows.index.to_frame())
```

Comment:
it shouldn't be necessary to do `rows.index.to_frame()` because the function will read directly from the index anyway